src/OrleansRuntime/Streams/QueueBalancer/DeploymentBasedQueueBalancer.cs

```diff
@@ -98,19 +98,33 @@ public class DeploymentBasedQueueBalancer : QueueBalancerBase, ISiloStatusListen
             await NotifyListeners();
         }
 
+        private async Task HandleSiloStatusChange()
+        {
+            while (!this.siloStatusOracle.CurrentStatus.IsTerminating())
+            {
+                try
+                {
+                    this.siloStatusSharedState = await this.siloStatusSharedState.NextAsync;
+                    HandleSiloStatusChange(this.siloStatusSharedState.State);
```

Comment:
And also keep a while loop here seems to bring more performance penalty, since there's always a thread occupied, keep running this while loop until `Terminating`. Am I right? 

So if this is the expected usage pattern, then we have a performance penalty here?